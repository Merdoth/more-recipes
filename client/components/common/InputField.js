import React from 'react';


const InputField = ({
  type, name, placeholder, error, value, onChange, label
}) => (
    <div
        className={ `form-group ${error ? 'has-error' : ''}` }
    >
        <label
            htmlFor= { name }
            className="control-label" >

            { label }

        </label>

        <input
            id=""
            value={ value }
            onChange={ onChange }
            type= { type }
            name= { name }
            className="form-control"
            placeholder= { placeholder }
            required=""
            autoFocus=""
        />

        <br />
    </div>
);

export default InputField;
