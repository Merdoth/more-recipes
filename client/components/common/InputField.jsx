import React from 'react';
import { PropTypes } from 'prop-types';

const InputField = ({
  type,
  name,
  placeholder,
  value,
  onChange,
  label,
  required
}) => (
  <div className="input-field buttonb">
    <label htmlFor={name} className="control-label">
      {label}
    </label>

    <input
      type={type}
      name={name}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className="form-control"
      required={required}
      style={{
        border: '0',
        outline: '0',
        background: 'transparent',
        borderBottom: '1px solid black',
        borderRadius: '3px'
      }}
    />
  </div>
);

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool.isRequired
};

InputField.defaultProps = {
  type: 'text'
};

export default InputField;
