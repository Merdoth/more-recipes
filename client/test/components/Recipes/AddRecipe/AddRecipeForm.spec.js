import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedAddRecipeForm, { AddRecipeForm } from
  '../../../../components/Recipes/AddRecipe/AddRecipeForm.jsx';
import InputField from '../../../../components/common/InputField.jsx';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

/* global jest */
const setup = () => {
  const props = {
    addRecipes: () => {},
    recipeReducer: {
      recipes: () => {}
    }
  };
  return shallow(<AddRecipeForm {...props} />);
};

describe('AddRecipeForm Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper.find(InputField)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  it('should add recipes to the state', () => {
    const event = {
      preventDefault: jest.fn(),
      addRecipes: jest.fn()
    };
    const wrapper = setup();
    const button = wrapper.find('.submitBtn');
    button.simulate('click', event);
  });
});

describe('onChange()', () => {
  it('should set recipeName to state when input values changes', () => {
    const event = {
      target: { name: 'recipeName', value: '' }
    };
    const wrapper = setup();
    const recipeNameInput = wrapper.find('#recipeName');

    event.target.value = 'Egusi Soup';
    recipeNameInput.simulate('change', event);

    expect(wrapper.instance().state.recipeName).toBe('Egusi Soup');
  });

  it('should set description state when input values changes', () => {
    const event = {
      target: { name: 'description', value: '' }
    };
    const wrapper = setup();
    const descriptionInput = wrapper.find('#description');

    event.target.value = 'This recipe is made with love';
    descriptionInput.simulate('change', event);

    expect(wrapper.instance().state.description).toBe('This recipe is made with love');
  });

  it('should set ingredients state when input values changes', () => {
    const event = {
      target: { name: 'ingredients', value: '' }
    };
    const wrapper = setup();
    const ingredientsInput = wrapper.find('#ingredients');

    event.target.value = 'pepper, salt, oil, water, fish';
    ingredientsInput.simulate('change', event);

    expect(wrapper.instance().state.ingredients).toBe('pepper, salt, oil, water, fish');
  });

  it('should set preparation state when input values changes', () => {
    const event = {
      target: { name: 'preparation', value: '' }
    };
    const wrapper = setup();
    const preparationInput = wrapper.find('#preparation');

    event.target.value = 'pepper, salt, oil, water, fish';
    preparationInput.simulate('change', event);

    expect(wrapper.instance().state.preparation).toBe('pepper, salt, oil, water, fish');
  });

  it('should set image state when input values changes', () => {
    const event = {
      preventDefault: jest.fn(),
      target: { name: 'image', value: '' }
    };
    const wrapper = setup();
    const imageInput = wrapper.find('#image');

    event.target.files = 'pepper.jpeg';
    imageInput.simulate('change', event);
    wrapper.setState({
      image: event.target.files
    });
    expect(wrapper.instance().state.image).toBe('pepper.jpeg');
  });
});

describe('Connected AddRecipeForm component', () => {
  it('tests that the component successfully rendered', () => {
    const store = mockStore({
      recipeReducer: {
        recipes: () => { }
      }
    });
    const wrapper = shallow(<ConnectedAddRecipeForm store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
