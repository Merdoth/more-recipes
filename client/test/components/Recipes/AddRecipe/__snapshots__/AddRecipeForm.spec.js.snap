// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipeForm Component snapshot it should render the right amount of elements 1`] = `
<form
  className="form-signin"
  onSubmit={[Function]}
>
  <h2
    className="form-signin-heading"
  >
    Add Recipe
  </h2>
  <hr />
  <InputField
    id="recipeName"
    label="Name"
    name="recipeName"
    onChange={[Function]}
    placeholder="Name"
    required={true}
    type="text"
    value=""
  />
  <TextArea
    id="description"
    label="Description"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <TextArea
    id="ingredients"
    label="Ingredients"
    name="ingredients"
    onChange={[Function]}
    placeholder="Ingredients"
    type="text"
    value=""
  />
  <TextArea
    id="preparation"
    label="Preparation"
    name="preparation"
    onChange={[Function]}
    placeholder="Preparation"
    type="text"
    value=""
  />
  <InputLine
    id="image"
    label="Select Image"
    name="image"
    onChange={[Function]}
    placeholder="Upload Image"
    type="file"
    value=""
  />
  <Button
    className="btn btn-lg btn-primary btn-block submitBtn"
    iconClass="fa-cutlery"
    name="Add Recipe"
    onClick={[Function]}
    type="submit"
  />
</form>
`;
